### PR TITLE
d_do_test: Enforce colon after test directives

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Because it should always indicate that it's a test parameter and not an unrelated token that contains it. For reference, the GDC test suite currently ignores test parameters without a colon.
    
Replaced the search for `GDB_SCRIPT` because shell scripts usually use  `GDB_SCRIPT=...` and the check doesn't require the actual content.

### CODE REVIEW

**Comment by ibuclaw** [comment]
> Because it should always indicate that it's a test directive and not an unrelated token that contains it. This is (partially?) enforced within the GDC test suite and should also be checked here.

Actually, the GDC test suite ignores it, because the Dejagnu scripts only match the directive _with_ colon.

i.e:
```
if [regexp -- {DFLAGS\s*:\s*(.*)} $line match args] {
 ...
} elseif [regexp -- {DISABLED:} $copy_line] {
 ...
}
 ```

**Comment by ibuclaw** [comment]
I'd prefer this to use words rather than symbols in the commit summary:
```
d_do_test: Enforce colon after test directives
```

**Comment by MoonlightSentinel** [comment]
> Actually, the GDC test suite ignores it, because the Dejagnu scripts only match the directive _with_ colon.

Thanks, I've corrected the commit summary accordingly.

> I'd prefer this to use words rather than symbols in the commit summary:

Sure

File: test/fail_compilation/invalid_lib.d [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
 REQUIRED_ARGS: -lib
-REQUIRED_ARGS(linux freebsd osx openbsd) fail_compilation/extra-files/fake.a
-REQUIRED_ARGS(windows) -m32 fail_compilation/extra-files/fake.lib
+REQUIRED_ARGS(linux freebsd osx openbsd): fail_compilation/extra-files/fake.a
+REQUIRED_ARGS(windows): -m32 fail_compilation/extra-files/fake.lib
```
Comment (WalterBright) [inline test/fail_compilation/invalid_lib.d:4]: Where did the (...) syntax come from? This is not documented.

File: test/tools/d_do_test.d [hunk 2]
```diff
@@ -318,6 +328,65 @@ bool findTestParameter(const ref EnvData envData, string file, string token, ref
...
+/*
+Here's a link
+FOO: a b
+FOO: c
+*/
+
+void main() {}
+
+// BAR(windows): all_win
+// BAR(windows64): 64_win
```
Comment (WalterBright) [inline test/tools/d_do_test.d:343]: `windows64` is not documented anywhere

File: test/fail_compilation/fail18417.d [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// REQUIRED_ARGS : -de
+// REQUIRED_ARGS: -de
```
Comment (WalterBright) [inline test/fail_compilation/fail18417.d:1]: If the space before : is allowed or not is not documented anywhere.

**Review by WalterBright — DISMISSED** [review]
It is not clear to me whether `XXTEST_OUTPUT:` is recognized as `TEST_OUTPUT` or not. (It shouldn't be.)

File: test/fail_compilation/invalid_lib.d (hunk 1 above)
Comment (MoonlightSentinel) [reply test/fail_compilation/invalid_lib.d:4]: Apparently missing from the README, will add in another PR.

File: test/tools/d_do_test.d (hunk 2 above)
Comment (MoonlightSentinel) [reply test/tools/d_do_test.d:343]: Same syntax as `DISABLED` values.

File: test/fail_compilation/fail18417.d (hunk 3 above)
Comment (MoonlightSentinel) [reply test/fail_compilation/fail18417.d:1]: Fixed

**Comment by MoonlightSentinel** [comment]
Adapted to the recent refactoring - so no more duplication for test / output parameters.

**Comment by RazvanN7** [comment]
@MoonlightSentinel is this ready for review?

**Comment by MoonlightSentinel** [comment]
Yes

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**